pythonforandroid/recipes/cppy/__init__.py

```diff
@@ -1,14 +1,10 @@
-from pythonforandroid.recipe import PythonRecipe
+from pythonforandroid.recipe import PyProjectRecipe
 
 
-class CppyRecipe(PythonRecipe):
+class CppyRecipe(PyProjectRecipe):
     site_packages_name = 'cppy'
-    version = '1.1.0'
-    url = 'https://github.com/nucleic/cppy/archive/{version}.zip'
-    call_hostpython_via_targetpython = False
-    # to be detected by the matplotlib install script
-    install_in_hostpython = True
-    depends = ['setuptools']
+    version = '08366d6fe4f385efde2dc0a38f95d7c3763db577'  # 1.2.1
+    url = 'git+https://github.com/nucleic/cppy'
```

Comment:
Why we're not just using `https://github.com/nucleic/cppy/archive/refs/tags/1.2.1.zip` or `https://github.com/nucleic/cppy/releases/download/1.2.1/cppy-1.2.1.tar.gz`?

Also, as it seems it's just needed for the `hostpython`, in case build-isolation is not supported, maybe we can just drop this recipe?

Reply:
Got it, but:
- If is just needed for hostpython dependency, we still need a recipe, or we can use `hostpython_prerequisites` ?
- We can't rely on the tag instead of commit hash for cloning ?

Reply:
Hm, looks like it just only provide build headers used by kiwisolver. I think it can be removed.

Reply:
Yeah, I think we actually have a lot of dependencies which are just meant to be installed in `hostpython` as we previously did not allowed installing things into hostpython environment, without a recipe.

With hostpython_prerequisites (and isolated builds, where applicable) these recipes can likely be removed.
(Don't be afraid of cleaning-up things 😅)